packages/cloudflare/src/cli/build/bundle-server.ts

```diff
@@ -164,6 +200,7 @@ async function updateWorkerBundledCode(
   patchedCode = await patchCache(patchedCode, openNextOptions);
   patchedCode = inlineMiddlewareManifestRequire(patchedCode, config);
   patchedCode = patchExceptionBubbling(patchedCode);
+  patchedCode = patchLoadInstrumentationModule(tsParseFile(patchedCode)).print();
```

Comment:
the `tsParseFile()` + `print()` looks a bit out of place here, but I think this is the appropriate direction actually, ideally we should call `tsParseFile()` at line 193, and use that `ts.SourceFile` for all the patching, and end with a single `print()`

Before we get to that, in my opinion, the more patches we can do here with AST the better.

(if someone disagree I am happy to keep going with the regex replacement)

Reply:
> move the parsing/stringify inside the function

I didn't want to do it inside the function since as I mentioned in my comment, I think that these functions should ideally all take an AST and return an AST, so actually following this structure already (without hiding the AST inside the function itself) make this clearer in my opinion, I am happy to move the parsing/stringifying into the function if you prefer that (it doesn't really make much of a difference in this case) 🙂

Reply:
updated as suggested 👍